TombEngine/Game/effects/effects.cpp

```diff
@@ -439,7 +439,172 @@ void TriggerCyborgSpark(int x, int y, int z, short xv, short yv, short zv)
 
 void TriggerExplosionSparks(int x, int y, int z, int extraTrig, int dynamic, int uw, int roomNumber)
 {
-	TriggerExplosion(Vector3(x, y, z), 512, true, false, true, roomNumber);
+	int dx, dz, mirror, i, scalar;
+	unsigned char extras_table[4];
+	unsigned char r, g, b;
+
+	extras_table[0] = 0;
+	extras_table[1] = 4;
+	extras_table[2] = 7;
+	extras_table[3] = 10;
+	dx = LaraItem->Pose.Position.x - x;
+	dz = LaraItem->Pose.Position.z - z;
+	scalar = 1;
+	mirror = 0;
+
+	if (dx < -0x4000 || dx > 0x4000 || dz < -0x4000 || dz > 0x4000)
```

Comment:
Can we replace 0x4000 with `BLOCK` macros?

Reply:
it was ANGLE(90.0f) for the other codes, so I took ANGLE. I even replaced the other 0x4000 (16.380) values in effects.cpp

Reply:
It's not right. `dx` and `dz` are position delta values, they have completely nothing to do with angles.